web/src/routes/admin/system-settings/+page.svelte

```diff
@@ -62,74 +65,195 @@
-      </SettingAccordion>
-
-      <SettingAccordion title="Map & GPS Settings" subtitle="Manage map related features and setting">
-        <MapSettings disabled={$featureFlags.configFile} config={configs} />
-      </SettingAccordion>
-
-      <SettingAccordion title="OAuth Authentication" subtitle="Manage the login with OAuth settings">
-        <OAuthSettings disabled={$featureFlags.configFile} oauthConfig={configs.oauth} />
-      </SettingAccordion>
-
-      <SettingAccordion title="Password Authentication" subtitle="Manage login with password settings">
-        <PasswordLoginSettings disabled={$featureFlags.configFile} passwordLoginConfig={configs.passwordLogin} />
-      </SettingAccordion>
-
-      <SettingAccordion title="Server Settings" subtitle="Manage server settings">
-        <ServerSettings disabled={$featureFlags.configFile} serverConfig={configs.server} />
-      </SettingAccordion>
-
-      <SettingAccordion
-        title="Storage Template"
-        subtitle="Manage the folder structure and file name of the upload asset"
-        isOpen={$page.url.searchParams.get('open') === 'storage-template'}
-      >
-        <StorageTemplateSettings disabled={$featureFlags.configFile} storageConfig={configs.storageTemplate} />
-      </SettingAccordion>
-
-      <SettingAccordion title="Theme Settings" subtitle="Manage customization of the Immich web interface">
-        <ThemeSettings disabled={$featureFlags.configFile} themeConfig={configs.theme} />
-      </SettingAccordion>
-
-      <SettingAccordion title="Thumbnail Settings" subtitle="Manage the resolution of thumbnail sizes">
-        <ThumbnailSettings disabled={$featureFlags.configFile} thumbnailConfig={configs.thumbnail} />
-      </SettingAccordion>
-
-      <SettingAccordion title="Trash Settings" subtitle="Manage trash settings">
-        <TrashSettings disabled={$featureFlags.configFile} trashConfig={configs.trash} />
-      </SettingAccordion>
-
-      <SettingAccordion title="Version Check" subtitle="Enable/disable the new version notification">
-        <NewVersionCheckSettings disabled={$featureFlags.configFile} newVersionCheckConfig={configs.newVersionCheck} />
-      </SettingAccordion>
-
-      <SettingAccordion
-        title="Video Transcoding Settings"
-        subtitle="Manage the resolution and encoding information of the video files"
-      >
-        <FFmpegSettings disabled={$featureFlags.configFile} ffmpegConfig={configs.ffmpeg} />
-      </SettingAccordion>
-    </section>
-  </section>
+  <AdminSettings {config} let:handleReset let:handleSave let:savedConfig let:defaultConfig>
+    <SettingAccordion title="Job Settings" subtitle="Manage job concurrency" isOpen={openSettings.includes('job')}>
+      <JobSettings
+        on:save={({ detail }) => handleSave(detail)}
+        on:reset={({ detail }) => handleReset(detail)}
+        disabled={$featureFlags.configFile}
+        {defaultConfig}
+        {config}
+        {savedConfig}
```

Comment:
You are pretty close to being able to change this whole thing for a for loop, where you have something like Array<{ title, subtitle, isOpen, component }> that you just loop over and only type out all these bindings once.

Reply:
That's the goal ;)

Reply:
Actually, I just got around to it so there you go! :)

Reply:
The settings page has still a few more lines than before... That bugs me lol